ui/src/pages/namespace/Permissions/components/PermisionsSelector/index.tsx

```diff
@@ -0,0 +1,128 @@
+  selectedPermissions: string[];
+  setPermissions: (permissions: string[]) => void;
+}) => {
+  const { t } = useTranslation();
+
+  const onCheckedChange = (permission: string, isChecked: boolean) => {
+    const currentPermissions = selectedPermissions;
+    const newPermissions = isChecked
+      ? [...currentPermissions, permission]
+      : currentPermissions.filter((p) => p !== permission);
+    setPermissions(newPermissions);
+  };
+
+  const allSelected =
+    selectedPermissions.length === availablePermissions?.length;
+  const noneSelected = selectedPermissions.length === 0;
+
+  const selectAllPermissions = () => {
+    setPermissions(availablePermissions);
+  };
+
+  const deselectAllPermissions = () => {
+    setPermissions([]);
+  };
+
+  const availableScopes = useMemo(
+    () => permissionStringsToScopes(availablePermissions),
+    [availablePermissions]
+  );
+
+  const groupedResouces = useMemo(
+    () => groupPermissionStringsByResouce(availablePermissions),
+    [availablePermissions]
+  );
+
+  return (
+    <>
+      <fieldset className="flex items-center gap-5">
+        <label className="w-[90px] text-right text-[14px]">
+          {t("pages.permissions.permissionsSelector.permissions")}
+        </label>
+
+        <Card className="max-h-[400px] w-full overflow-scroll" noShadow>
+          <Table>
+            <TableHead>
+              <TableRow className="hover:bg-inherit dark:hover:bg-inherit">
+                <TableHeaderCell sticky className="px-0">
+                  <div className="flex gap-5">
+                    <Button
+                      variant="link"
+                      type="button"
+                      size="sm"
+                      onClick={selectAllPermissions}
+                      disabled={allSelected}
+                    >
+                      <MousePointerSquare />
+                      {t("pages.permissions.permissionsSelector.selectAll")}
+                    </Button>
+                    <Button
+                      variant="link"
+                      size="sm"
+                      type="button"
+                      onClick={deselectAllPermissions}
+                      disabled={noneSelected}
+                    >
+                      <MousePointerSquareDashed />
+                      {t("pages.permissions.permissionsSelector.deselectAll")}
+                    </Button>
+                  </div>
+                </TableHeaderCell>
+                {availableScopes.map((scope) => (
+                  <TableHeaderCell
+                    sticky
+                    key={scope}
+                    className="w-20 px-2 text-center"
+                  >
+                    {scope.toLowerCase()}
+                  </TableHeaderCell>
+                ))}
+              </TableRow>
+            </TableHead>
+            <TableBody>
+              {Object.entries(groupedResouces)
+                .sort((a, b) => a[0].localeCompare(b[0]))
```

Comment:
This could be more readable if we defined a `const sortedResources` before the return statement, and only map over an array here without extra logic.

Please also note the r is missing in `groupedResouces` and other variable names as well.

Reply:
👌 c18e2590c1969ab7cc64527ae46ae4cdf2bda303